### PR TITLE
Don't use sed in _tc_activation

### PR DESCRIPTION
**Pull request by manopapad**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)

The clang activation script causes conda activation to take ~5s on my Mac laptop, which is quite annoying for those of us who want to `conda activate` on shell startup. The reason for this is that `sed` is called in a loop in `_tc_activation`, to split a string at the first occurrence of a comma, and `sed` seems to be quite slow on Mac. However, there is no reason to use `sed` for this; substring-substitution is a fast and portable way to achieve the same effect.

This change doesn't apply to just this repo's recipe, it could be copied onto every activation script that uses `_tc_activation`, but I couldn't easily find a good place to make this change globally.

### CODE REVIEW

**Comment by manopapad** [comment]
@conda-forge-admin, please rerender

**Comment by LunarLanding** [comment]
+1, takes 2s just with the 'compilers' package installed.
Seems there are conflicts, I guess the original recipe diverged from these changes?

**Comment by h-vetinari** [comment]
@isuruf, I rebased this old PR based on user feedback. To me the sed-replacements seem more involved than what's happening in the substring substitution (at least in principle), and since I'm not familiar with this part, I'd appreciate your input here.

File: recipe/deactivate-clang++.sh [hunk 1]
```diff
@@ -49,8 +49,8 @@ function _tc_activation() {
     for thing in "$@"; do
       case "${thing}" in
         *,*)
-          newval=$(echo "${thing}" | sed "s,^[^\,]*\,\(.*\),\1,")
-          thing=$(echo "${thing}" | sed "s,^\([^\,]*\)\,.*,\1,")
+          newval="${thing#*,}"
+          thing="${thing%%,*}"
```
Comment (isuruf) [inline recipe/deactivate-clang++.sh:53]: For MacOS we neeed
1. Bash 3 compat
2. Bash 4 compat
3. Zsh compat

For Linux we need
1. POSIX sh compat

I don't think this is POSIX sh compatible.

Comment (LunarLanding) [reply recipe/deactivate-clang++.sh:53]: It seems string substitution is POSIX (see references), and that string substitution matches sed in a linux machine (see test script).
The string substitution used works like this:
	•	`${parameter#word}`: Remove the smallest prefix pattern.
	•	`${parameter%%word}`: Remove the largest suffix pattern.
References:
[Stackoverflow](https://unix.stackexchange.com/a/389659) 
[POSIX Shell Command Language - Parameter Expansion](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02)

Test script (not output is good)
```sh
#!/bin/sh

test_cases='
""                # Empty string
","               # Single comma
",value1"         # Leading comma
"value1,"         # Trailing comma
"value1"          # Single value
"value1,value2"   # Two values
"value1,,value3"  # Consecutive commas
",,,"             # Only commas
"value1,value2,value3" # Multiple values
"value1,value2,value3,value4"
"value1,value2,,value4"
"value1,,value3,value4"
"value1,,,"
",,,"
"value1,value2,value3,value4,value5,value6"
'

# Remove comments and empty lines
test_cases=$(printf '%s\n' "$test_cases" | sed '/^\s*#/d;/^\s*$/d')

# Loop over each test case
echo "$test_cases" | while IFS= read -r line; do
    thing=$(printf '%s' "$line" | sed 's/^"//;s/"$//') # Remove surrounding quotes

    # Original sed method

    newval_sed=$(echo "${thing}" | sed "s,^[^\,]*\,\(.*\),\1,")
    thing_sed=$(echo "${thing}" | sed "s,^\([^\,]*\)\,.*,\1,")

    # Parameter expansion method
    newval_param="${thing#*,}"
    thing_param="${thing%%,*}"

    # Check if outputs match
    if [ "$newval_sed" = "$newval_param" ] || [ "$thing_sed" = "$thing_param" ]; then
         # echo "Match for input: '$thing'"
         # echo "  Output: thing='$thing_param', newval='$newval_param'"
    else
	echo "Mismatch for input: '$thing'"
        echo "  sed method:      thing='$thing_sed', newval='$newval_sed'"
        echo "  param expansion: thing='$thing_param', newval='$newval_param'"
    fi
done

```

I also captured the values of `$thing` for both a osx and a linux installation , files attached, and used the same test program (write instead `test_cases=$(cat things.txt)`), with no mismatches.
[things_linux.txt](https://github.com/user-attachments/files/17804156/things_linux.txt)
[things_osx.txt](https://github.com/user-attachments/files/17804158/things_osx.txt)

File: recipe/meta.yaml [hunk 2]
```diff
@@ -6,7 +6,7 @@
 # of the C++ stdlib; keep the variable in case this happens again
 {% set libcxx_major = major_ver %}
 
-{% set build_number = 21 %}
+{% set build_number = 22 %}
```
Comment (h-vetinari) [inline recipe/meta.yaml:9]: Main is already on 22
https://github.com/conda-forge/clang-compiler-activation-feedstock/blob/d8ea0127a69042f122d94299c2948bdaafac1707/recipe/meta.yaml#L9
so this needs another bump

**Review by h-vetinari — APPROVED** [review]
Thanks for taking care of this! Lgtm aside from a build number bump

File: recipe/activate-clang.sh [hunk 3]
```diff
@@ -170,9 +170,26 @@ _tc_activation \
   activate @CHOST@- "HOST,@CHOST@" \
   "CONDA_TOOLCHAIN_HOST,@CHOST@" \
   "CONDA_TOOLCHAIN_BUILD,@CBUILD@" \
-  ar as checksyms install_name_tool libtool lipo nm nmedit otool \
-  pagestuff ranlib redo_prebinding seg_addr_table seg_hack segedit size strings strip \
-  clang ld \
+  "AR,${AR:-@CHOST@-ar}" \
```
Comment (isuruf) [inline recipe/activate-clang.sh:173]: This needs to be ported to deactivate too

**Pull request merged**